test/OpenTelemetry.Tests/Logs/LogRecordSharedPoolTests.cs

```diff
@@ -45,7 +45,7 @@ public void RentReturnTests()
         Assert.Equal(1, pool.Count);
 
         // Note: This is ignored because logRecord manually created has PoolReferenceCount = int.MaxValue.
-        LogRecord manualRecord = new();
+        LogRecord manualRecord = new() { Source = LogRecord.LogRecordSource.FromSharedPool };
```

Comment:
Is this a valid scenario? What will be the scenario when logRecord created manually will have source as `FromSharedPool`

Reply:
I tweaked the test to be a little more realistic.